addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,554 @@
+        Connection currentConnection = connection;
+        Gson gson = new Gson();
+        JsonFeed[] feeds = gson.fromJson(flashBriefingJson, JsonFeed[].class);
+        if (currentConnection != null) {
+            try {
+                currentConnection.setEnabledFlashBriefings(feeds);
+            } catch (IOException | URISyntaxException e) {
+                logger.warn("Set flashbriefing profile failed {}", e);
+            }
+        }
+        updateFlashBriefingHandlers();
+    }
+
+    public void updateFlashBriefingHandlers() {
+        Connection currentConnection = connection;
+        if (currentConnection != null) {
+            updateFlashBriefingHandlers(currentConnection);
+        }
+    }
+
+    private void updateFlashBriefingHandlers(Connection currentConnection) {
+        synchronized (flashBriefingProfileHandlers) {
+            if (!flashBriefingProfileHandlers.isEmpty() || currentFlashBriefingJson.isEmpty()) {
+                updateFlashBriefingProfiles(currentConnection);
+            }
+
+            boolean flashBriefingProfileFound = false;
+            for (FlashBriefingProfileHandler child : flashBriefingProfileHandlers) {
+                flashBriefingProfileFound |= child.initialize(this, currentFlashBriefingJson);
+            }
+            if (flashBriefingProfileHandlers.isEmpty()) {
+                discoverFlashProfiles = true; // discover at least one device
+            }
+            if (discoverFlashProfiles) {
+                discoverFlashProfiles = false;
+                if (!flashBriefingProfileFound) {
+                    amazonEchoDiscovery.discoverFlashBriefingProfiles(getThing().getUID(),
+                            this.currentFlashBriefingJson, this.flashBriefingProfileHandlers.size() + 1);
+                }
+            }
+        }
+    }
+
+    public @Nullable Connection findConnection() {
+        return this.connection;
+    }
+
+    public String getEnabledFlashBriefingsJson() {
+        Connection currentConnection = this.connection;
+        if (currentConnection == null) {
+            return "";
+        }
+        updateFlashBriefingProfiles(currentConnection);
+        return this.currentFlashBriefingJson;
+    }
+
+    private void updateFlashBriefingProfiles(Connection currentConnection) {
+        try {
+            JsonFeed[] feeds = currentConnection.getEnabledFlashBriefings();
+            // Make a copy and remove changeable parts
+            JsonFeed[] forSerializer = new JsonFeed[feeds.length];
+            for (int i = 0; i < feeds.length; i++) {
+                JsonFeed source = feeds[i];
+                JsonFeed copy = new JsonFeed();
+                copy.feedId = source.feedId;
+                copy.skillId = source.skillId;
+                // Do not copy imageUrl here, because it will change
+                forSerializer[i] = copy;
+            }
+            Gson gson = new Gson();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.